app/controllers/traderDetails/consignment/CheckYourAnswersController.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.traderDetails.consignment
+
+import com.google.inject.Inject
+import controllers.actions.Actions
+import models.LocalReferenceNumber
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import viewModels.traderDetails.TraderDetailsConsignmentViewModel.TraderDetailsConsignmentSubSectionViewModel
+import views.html.traderDetails.consignment.CheckYourAnswersView
+
+class CheckYourAnswersController @Inject() (
+  override val messagesApi: MessagesApi,
+  actions: Actions,
+  val controllerComponents: MessagesControllerComponents,
+  view: CheckYourAnswersView,
+  viewModel: TraderDetailsConsignmentSubSectionViewModel
+)() extends FrontendBaseController
+    with I18nSupport {
+
+  def onPageLoad(lrn: LocalReferenceNumber): Action[AnyContent] = actions.requireData(lrn) {
+    implicit request =>
+      val sections = viewModel(request.userAnswers)
+      Ok(view(lrn, sections))
+  }
+
+  def onSubmit(lrn: LocalReferenceNumber): Action[AnyContent] = actions.requireData(lrn) {
+    Redirect(controllers.routes.TaskListController.onPageLoad(lrn))
```

Comment:
Should this not redirect to the trader details CYA?

Reply:
Note: The Trader Details CYA is the page that has all the data on from all the sections and is used when you want the edit the entire Trader Details section. So it would need a different Trader Details Summary CYA at best

Reply:
Right ok I'm with you. Yeah I guess that would work. So you only hit the 'summary' page if you click the 'Edit trader details' link. Happy to approve for now, but worth leaving a note for QAs as this goes against the journey flow diagram